libp2p/protocols/connectivity/autonatv2/types.nim

```diff
@@ -10,25 +10,38 @@
 {.push raises: [].}
 
 import results, chronos, chronicles
-import ../../../multiaddress, ../../../peerid #, ../../../errors
-import ../../../protobuf/minprotobuf
+import
+  ../../../multiaddress,
+  ../../../peerid,
+  ../../../protobuf/minprotobuf,
+  ../../../switch,
+  ../autonat/service
```

Comment:
Probably makes more sense to extract from service the things that are common to autonat v1 and v2 into some file that's shared by both that file and this one

Reply:
I wonder where that file would live
1. in `libp2p/protocols/connectivity/file.nim`
2. in `libp2p/protocols/connectivity/autonat/file.nim` (in this case `autonatv2/` should live inside `autonat/`, and `autonatv1` should also be inside `autonat/`)
3. Same as 2 but without creating a new `autonatv1` folder. Which is the same as just renaming `core.nim` to `types.nim` and importing it in v2.

I'll go with 3: change `core.nim` to `types.nim` in `autonat/` and import `autonat/types` in v2.